### PR TITLE
AP-4610: Update partner means report

### PR DESCRIPTION
**Pull request by colinbruce**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-4610)

Update the means report to ensure that the data is presented in the best way for caseworkers, splitting headings to client/partner and with the requested grouping

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

**Comment by colinbruce** [comment]
@naseberry I assume we are okay to ignore sonarcloud?  It doesn't like the `<dt>` tags in a partial not being surrounded by a `<dl>` tag but that is in the parent page.  It renders like this:
image

Also, just spotted duplicate IDs so I'll get those refactored! 😬

**Comment by naseberry** [comment]
> @naseberry I assume we are okay to ignore sonarcloud? It doesn't like the `<dt>` tags in a partial not being surrounded by a `<dl>` tag but that is in the parent page. It renders like this: image
> 
> Also, just spotted duplicate IDs so I'll get those refactored! 😬

Yes, in this case it can be ignored in SonarCloud.

File: app/views/shared/means_report/_item.html.erb [hunk 1]
```diff
@@ -1,12 +1,25 @@
 <% addendum = local_assigns[:addendum] || nil %>
 <% suppress_border ||= false %>
 <% css_border_klass = suppress_border ? "govuk-summary-list__row--no-border" : "" %>
-<div class="<%= "govuk-summary-list__row  normal-word-break #{css_border_klass}" %>" id="<%= "means-merits-report__#{name}" %>">
+<div class="<%= "govuk-summary-list__row  normal-word-break" %>" id="<%= "means-merits-report__#{name}" %>">
```
Comment (naseberry) [inline app/views/shared/means_report/_item.html.erb:4]: nit: `class="<%= "govuk-summary-list__row normal-word-break" %>"` are the erb opening and closing tags required or can they be removed?

File: app/views/shared/means_report/_item.html.erb [hunk 2]
```diff
@@ -1,12 +1,25 @@
...
   <dt class="<%= "govuk-summary-list__key govuk-!-width-one-half" %>">
```
Comment (naseberry) [inline app/views/shared/means_report/_item.html.erb:5]: ditto on erb tags

File: config/locales/en/shared.yml [hunk 3]
```diff
@@ -213,6 +213,11 @@ en:
         deductions:
           dependants_allowance: Dependants allowance
           disregarded_state_benefits: Income from benefits excluded from calculation
+          partner_allowance: Partner Allowance
```
Comment (naseberry) [inline config/locales/en/shared.yml:216]: nit: following the existing content case types, should `Partner Allowance` be `Partner allowance`

**Review by naseberry — APPROVED** [review]

**Pull request merged**